Sources/ApolloCodegenLib/Frontend/CompilationResult.swift

```diff
@@ -0,0 +1,184 @@
+import JavaScriptCore
+
+/// The output of the frontend compiler.
+public struct CompilationResult: Decodable {
+  var operations: [OperationDefinition]
+  var fragments: [FragmentDefinition]
+  var typesUsed: [GraphQLNamedType]
+  
+  public struct OperationDefinition: Decodable, CustomStringConvertible {
+    var operationName: String
+    var operationType: OperationType
+    
+    var variables: [VariableDefinition]
+    
+    var rootType: GraphQLCompositeType
+    var selectionSet: SelectionSet
+    
+    var source: String
+    var filePath: String
+    
+    var operationIdentifier: String {
+      // TODO: Compute this from source + referenced fragments
+      fatalError()
```

Comment:
^this still needs to have some kind of resolution

Reply:
This isn't actually part of the GraphQL spec, but part of our our own support for automatic persisted queries. The code in our old codegen that computes this is [here](https://github.com/apollographql/apollo-tooling/blob/master/packages/apollo-codegen-core/src/compiler/visitors/generateOperationId.ts). (We can't use `crypto` from JavaScriptCode, so we need to compute the SHA from Swift.)

Reply:
Also, these are the [tests](https://github.com/apollographql/apollo-tooling/blob/master/packages/apollo-codegen-core/src/compiler/visitors/__tests__/generateOperationId.ts) for this, so we'd probably want to port those to Swift as well.

Reply:
Thanks @martijnwalraven . I'm confused on why we need to port this to Swift. Couldn't the IR compiler generate the object identifiers and spit them out. Seems like it should be part of the IR compilers job so that if we ever want to use the compiler with other languages that functionality it already supported.

Reply:
I think we can deal with this in a follow-on PR, this one's gone on long enough